File: Python/chapitre7/Exercice1.py
import os
import signal
import sys
pid = os.fork()
if pid == 0:
    while True:
        pass
while True:
    k = sys.stdin.read(1)
    if k == 's':
        os.kill(pid, signal.SIGSTOP)
    elif k == 'r':
        os.kill(pid, signal.SIGCONT)
    elif k == 'q':
        os.kill(pid, signal.SIGTERM)
        break
os.wait()

File: C/chapitre7/Exercice1.c
#include <stdio.h>
#include <unistd.h>
#include <signal.h>
#include <sys/wait.h>

int main(void) {
    pid_t pid = fork();
    if (pid == 0) {
        while (1) { }
    }
    char k;
    while (scanf(" %c", &k) == 1) {
        if (k == 's') kill(pid, SIGSTOP);
        else if (k == 'r') kill(pid, SIGCONT);
        else if (k == 'q') {
            kill(pid, SIGTERM);
            break;
        }
    }
    wait(NULL);
    return 0;
}